ext/geos_c_impl/geometry.c

```diff
@@ -1042,18 +1042,38 @@ static VALUE method_geometry_invalid_reason(VALUE self)
   self_data = RGEO_GEOMETRY_DATA_PTR(self);
   self_geom = self_data->geom;
   if (self_geom) {
-    str = GEOSisValidReason_r(self_data->geos_context, self_geom);
-    // Per documentation, a valid geometry should give an empty string.
-    // However it seems not to be the case. Hence the comparison against
-    // the string that is really given: `"Valid Geometry"`.
-    // See https://github.com/libgeos/geos/issues/431.
-    if (str) result = (str[0] == '\0' || !strcmp(str, "Valid Geometry")) ? Qnil : rb_str_new2(str);
-    else result = rb_str_new2("Exception");
-    GEOSFree_r(self_data->geos_context, str);
+    // TODO: should we consider using the flag GEOSVALID_ALLOW_SELFTOUCHING_RING_FORMING_HOLE?
```

Comment:
@keithdoggett since you implemented validity on the ruby part, may I have your opinion on that one?

Reply:
Well since users never had this option and no one requested it I'm not sure I'm in favor of that. The question was more : we have to pick one option (either always or never allow), which is the ruby counterpart? I'll try and find an example in the GEOS test suite to answer that question myself

Reply:
```ruby
def ring(pts, factory: RGeo::Geos.factory)
  pts = pts.each_slice(2) if pts.first.is_a?(Numeric) && pts.size.even?
  factory.linear_ring(
    pts.map { factory.point(_1, _2) }
  )
end
[RGeo::Geos.factory, RGeo::Cartesian.simple_factory].each do |f|
	puts ring([0,0,	5,5,	0,10,	10,10,	5,5,	10,0,	0,0], factory: f).invalid_reason
end
```

Hence I'd be in favor of not adding the flag and sticking to SFS.

Another thing I found however that we may want to fix before releasing v3 is that when calling `make_valid` on the above ring we get `nil`. As we want to harmonize and always raise on error, I'd we should raise in that particular case. What do you say?

Reply:
Ahh, gotcha yes, I agree that we should not use that flag and the ring self-intersections be treated as invalid since that is not standard.

Good catch with `make_valid` return `nil`. Should definitely raise if it can't make it valid.